Webterminal: Added SSL support with existing LE certificates.

ludeeus opened the pull request:
## Description:
This add the option to use existing Let's Encrypt certificates.

Tested with:
- [X] Generated with Dehydrated. (DuckDNS script) tested by @ludeeus
- [X] Generated with certbot. _Simulated testing bt @ludeeus, se comment, this should work._

**Related issue (if applicable):** Fixes #74 

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Script has validation check of the job.
  
### If pertinent:
  - [X] Created/Updated documentation at `/docs`
  
![image](https://user-images.githubusercontent.com/15093472/36945814-b0e96638-1fb3-11e8-974e-e6096691069b.png)

ludeeus committed:
Added SSL support with existig LE certs

ludeeus committed:
Fixes for sticker-ci

ludeeus committed:
Updated documentation.

ludeeus commented:
@juan11perez Can you test this with your certificates generated with the certbot?

Landrash commented on docs/webterminal.md:
```diff
@@ -27,5 +27,17 @@ panel_iframe:
+#### Notes for SSL
+If you enable the use of existing Let's Encrypt certificates you need to open ports in your firewall to use them.
+
+If SSL is used the panel_iframe has to use your domain.
```
Use **the same** domain.

Landrash commented on package/opt/hassbian/suites/files/webterminalsslhelper.sh:
```diff
@@ -0,0 +1,16 @@
+FULLCHAIN=$(find "$CERTDIR" -type f | grep fullchain)
+PRIVKEY=$(find "$CERTDIR" -type f | grep privkey)
+DOMAIN=$(ls "$CERTDIR")
+cat "$FULLCHAIN" "$PRIVKEY" > /var/lib/shellinabox/certificate-"$DOMAIN".pem
```
Doing it this way would not link the cert but only "copy" it. Preferable would be to create a symlink to the certificate so that it's linked to the used certificate.

ludeeus replied:
In every guide I have found, and my own testing. the cert file used for shellinabox has to include both certificate and privkey, is it possible to merge with linking?

Landrash replied:
I missed that. That would require some strange magic.
Most likely this will stop working as soon as the certificate expires. If your ok with that then we can merge this.
Could you add to the update function to update the certificate if it's there?

ludeeus replied:
I hope not :( that defeats the purpose of the separate webterminalhelper.sh script..
Should't [this](https://github.com/home-assistant/hassbian-scripts/pull/130/files#diff-f9e7a0ede4e3162455ffa1299a2f1a19R52) update the cert for shellinabox 1AM every night?
looking at the timestamp after running the file it looks like that would work.
![image](https://user-images.githubusercontent.com/15093472/37119699-7dd93a14-2258-11e8-925c-53bc273b67ed.png)
I can add a note about running that file in the docs?

"Could you add to the update function to update the certificate if it's there?"
Do you mean the update for dehydrated with the DuckDNS script, then no. that will result in certbot certs not being updated.
If this was not what you ment, enlighten me :D

Landrash replied:
I missed that part of the webterminalhepler.sh script.

This look great now!

Landrash reviewed (changes requested):
Question is if we want to copy the certificate or if you should try to link the certificate instead.

ludeeus committed:
Specified which domain to use.

ludeeus committed:
Added correct protocol

ludeeus committed:
Updated cronjob

Landrash reviewed (dismissed):
Looks good to me. 
Ready to merge? If so remove the WIP tag :)

ludeeus commented:
I still want someone that can use certbot certs to verify this before er merge.

juan11perez commented:
@ludeeus  i'll pull out my Pi and test it. I still have a valid installation, I think ;-)
where do I get the script from ?

ludeeus commented:
@juan11perez Great!
Its in this PR, but hit me up on discord and I can help you :)

juan11perez commented:
@ludeeus ok, give me about half hour and ill contact you in discord

ludeeus commented:
Steps for testing this:
First upgrade hassbian-config to dev:
```bash
sudo hassbian-config upgrade hassbian-script && sudo hassbian-config upgrade hassbian-script-dev
```
Then download the new files:
```bash
sudo curl -o /opt/hassbian/suites/webterminal.sh https://raw.githubusercontent.com/ludeeus/hassbian-scripts/webterminal-ssl/package/opt/hassbian/suites/webterminal.sh && sudo curl -o /opt/hassbian/suites/files/webterminalhelper.sh https://raw.githubusercontent.com/ludeeus/hassbian-scripts/webterminal-ssl/package/opt/hassbian/suites/files/webterminalsslhelper.sh
```
Then run it:
```bash
sudo hassbian-config install webterminal --force
```

If you at an earlier time have installed webterminal, you need to change the settings for it:
`sudo nano /etc/default/shellinabox`
Change `SHELLINABOX_ARGS="--no-beep --disable-ssl"` to `SHELLINABOX_ARGS="--no-beep"`

ludeeus committed:
Delayed update part 5 min

juan11perez commented:
@ludeeus not working. see entire process
https://hastebin.com/xuwajoheci.erl

ludeeus committed:
Changed logic for setting privkey and  fullchain

ludeeus committed:
changed restart to sto start

ludeeus commented:
I added my fulchain.pem and privkey.pem to an folder named `/etc/letsencrypt/live/mydomain.com/`
It ran the script and used those cert files.
This make me belive that this will work with certbot certs also.
@Landrash I have now removed WIP, can you take a check on this now? :D

Landrash reviewed (approved):
Looks good to me.
Great job!
